deploy/templates/nstemplatetiers/appstudio-env/ns_env.yaml

```diff
@@ -0,0 +1,169 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: appstudio-env
+objects:
+- apiVersion: v1
+  kind: Namespace
+  metadata:
+    annotations:
+      openshift.io/description: ${USERNAME}
+      openshift.io/display-name: ${USERNAME}
+      openshift.io/requester: ${USERNAME}
+    labels:
+      name: ${USERNAME}-env
+      toolchain.dev.openshift.com/workspace: ${USERNAME}
+      argocd.argoproj.io/managed-by: gitops-service-argocd
+    name: ${USERNAME}-env
+
+# ServiceAccount to be used by the deployer (eg. ArgoCD) to manage the content of the namespace
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    namespace: ${USERNAME}-env
+    name: namespace-manager
+
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: RoleBinding
+  metadata:
+    name: namespace-manager
+    namespace: ${USERNAME}-env
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: edit
+  subjects:
+  - kind: ServiceAccount
+    name: namespace-manager
+
+# Quotas and default limits for not-terminating containers (regular long-running containers)
+- apiVersion: v1
+  kind: ResourceQuota
+  metadata:
+    name: compute-deploy
+    namespace: ${USERNAME}-env
+  spec:
+    scopes:
+      - NotTerminating
```

Comment:
We also have to add a quota for terminating containers. My understanding is if we do not specify it it means there is no limitation for terminating containers. Let's add a quota for terminating containers but let's set it to "0" (would need to test if "0" actually means "0" and not something like "no-limits :-)

Reply:
good point. 
replaced with c450d404
and tested locally:
```$ k describe quota -n testingtiers-env
Name:       compute-build
Namespace:  testingtiers-env
Scopes:     Terminating
 * Matches all pods that have an active deadline. These pods have a limited lifespan on a node before being actively terminated by the system.
Resource  Used  Hard
--------  ----  ----
cpu       0     0
memory    0     0
pods      0     0


Name:       compute-deploy
Namespace:  testingtiers-env
Scopes:     NotTerminating
 * Matches all pods that do not have an active deadline. These pods usually include long running pods whose container command is not expected to terminate.
Resource         Used  Hard
--------         ----  ----
limits.cpu       0     20
limits.memory    0     32Gi
requests.cpu     0     1750m
requests.memory  0     32Gi


Name:                         storage
Namespace:                    testingtiers-env
Resource                      Used  Hard
--------                      ----  ----
count/persistentvolumeclaims  0     12
limits.ephemeral-storage      0     50Gi
requests.ephemeral-storage    0     50Gi
requests.storage              0     50Gi
```
if no quota was specified then it was:
```
Name:       compute-deploy
Namespace:  testingtiers-env
Scopes:     NotTerminating
 * Matches all pods that do not have an active deadline. These pods usually include long running pods whose container command is not expected to terminate.
Resource         Used  Hard
--------         ----  ----
limits.cpu       0     20
limits.memory    0     32Gi
requests.cpu     0     1750m
requests.memory  0     32Gi


Name:                         storage
Namespace:                    testingtiers-env
Resource                      Used  Hard
--------                      ----  ----
count/persistentvolumeclaims  0     12
limits.ephemeral-storage      0     50Gi
requests.ephemeral-storage    0     50Gi
requests.storage              0     50Gi

```